en/docs/install-and-setup/setup/distributed-deployment/understanding-the-distributed-deployment-of-wso2-api-m.md

```diff
@@ -91,3 +91,238 @@ In the following diagram, the five components are set up in a distributed deploy
+        <td>
+            Authenticate to publishing events
+        </td>
+    </tr>
+    <tr class="odd">
+    <td>Publisher</td>
+    <td>Gateway</td>
+    <td>9443</td>
+        <td>
+           Deploy API artifacts
+        </td>
+    </tr>
+    <tr class="even">
+    <td>Publisher</td>
+    <td>Analytics Worker</td>
+    <td>7612</td>
+         <td>
+            Publishing alert analytics events
+         </td>
+    </tr>
+    <tr class="odd">
+    <td>Publisher</td>
+    <td>Analytics Worker</td>
+    <td>7712</td>
+        <td>
+            Authenticate to publishing alert analytics events
+        </td>
+    </tr>
+    <tr class="even">
+    <td>Publisher</td>
+    <td>Analytics Worker</td>
+    <td>7444</td>
+        <td>
+            Fetch all the generated alerts
+        </td>
+    </tr>
+    <tr class="odd">
+    <td>Dev Portal</td>
+    <td>Key Manager</td>
+    <td>9443</td>
+         <td>
+            Register auth app and generate app keys
+         </td>
+    </tr>
+    <tr class="even">
+    <td>Dev Portal</td>
+    <td>Gateway</td>
+    <td>8243</td>
+         <td>
+            Token endpoint and revoke endpoint
+         </td>
+    </tr>
+    <tr class="odd">
+    <td>Dev Portal</td>
+    <td>Analytics Worker</td>
+    <td>7612</td>
+    <td>
+        Publishing alert analytics events
+    </td>
+    </tr>
+    <tr class="even">
+    <td>Dev Portal</td>
+    <td>Analytics Worker</td>
+    <td>7712</td>
+        <td>
+           Authenticate to publishing alert analytics events
+        </td>
+    </tr>
+    <tr class="odd">
+    <td>Analytics Dashboard</td>
+    <td>Analytics Worker</td>
+    <td>9444</td>
+         <td>
+            Deploy business rules
+         </td>
+    </tr>
+    <tr class="even">
+    <td>Analytics Dashboard</td>
+    <td>Publisher</td>
+    <td>9443</td>
+         <td>
+            Use admin and Publisher rest APIs
+         </td>
+    </tr>
+    <tr class="odd">
+    <td>Analytics Dashboard</td>
+    <td>Dev Portal</td>
+    <td>9443</td>
+         <td>
+            Use Dev Portal portal rest APIs
+         </td>
+    </tr>
+    </tbody>
+    </table>
+    **Open port for Load balancer/User**
+    <table>
+    <thead>
+    <tr class="header">
+    <th>Server</th>
+    <th>Port</th>
+    <th>Description</th>
+    </tr>
+    </thead>
+    <tbody>
+    <tr class="odd">
+    <td>Key Manager</td>
+    <td>9443</td>
+        <td>
+            Token endpoint and identity rest APIs/web apps
+        </td>
+    </tr>
+    <tr class="even">
+    <td>Gateway</td>
+    <td>8243</td>
+        <td>
+            Gateway nio HTTPS
+        </td>
+     </tr>
+    <tr class="odd">
+    <td>Gateway</td>
+    <td>8280</td>
+        <td>
+            Gateway nio HTTP
+        </td>
+    </tr>
+    <tr class="even">
+    <td>Gateway</td>
+    <td>8099</td>
+        <td>
+            Gateway WS
+        </td>
+    </tr>
+    <tr class="odd">
+    <td>Gateway</td>
+    <td>9099</td>
+        <td>
+            Gateway WSS
```

Comment:
WSS / WS - Shall we include web socket/ web socket secure in brackest

Reply:
Fixed https://github.com/wso2/docs-apim/pull/2794/commits/53f2b25ff98ad62ce7f3f97ec7273e932e4825b0